src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -655,10 +656,23 @@ def _save_checkpoint(
 
         # COMPUTE THE CURRENT metric
         # IF idx IS A LIST - SUM ALL THE VALUES STORED IN THE LIST'S INDICES
-        metric = validation_results_dict[self.metric_to_watch]
+        curr_tracked_metric = validation_results_dict[self.metric_to_watch]
+
+        # create metrics dict to save
+        valid_metrics_titles = get_metrics_titles(self.valid_metrics)
+
+        all_metrics = {
+            "tracked_metric_name": self.metric_to_watch,
+            "metrics": {"valid": {metric_name: float(validation_results_dict[metric_name]) for metric_name in valid_metrics_titles}},
+        }
+
+        train_metrics_titles = get_metrics_titles(self.train_metrics)
+
+        if train_metrics_dict is not None:
+            all_metrics["metrics"]["train"] = {metric_name: float(train_metrics_dict[metric_name]) for metric_name in train_metrics_titles}
 
         # BUILD THE state_dict
-        state = {"net": unwrap_model(self.net).state_dict(), "acc": metric, "epoch": epoch}
+        state = {"net": unwrap_model(self.net).state_dict(), "acc": curr_tracked_metric, "epoch": epoch, "all_metrics": all_metrics}
```

Comment:
The final checkpoint would contain: ["all_metrics"]["metrics"]["train"]
A all_metrics/metrics seems a bit 'extra' no?